pkg/store/bucket.go

```diff
@@ -746,6 +751,7 @@ func (b *bucketBlock) readIndexRange(ctx context.Context, off, length int64) ([]
 	}
 	defer r.Close()
 
+	// NOTE(bplotka): Huge amount of memory is allocated here. We need to cache it.
```

Comment:
Initially we could address heavy allocs via simple memory pooling instead of actually caching data.